src/vm/comutilnative.cpp

```diff
@@ -2624,14 +2626,107 @@ FCIMPL6(INT32, ManagedLoggingHelper::GetRegistryLoggingValues, CLR_BOOL* bLoggin
 }
 FCIMPLEND
 
+static BOOL HasOverriddenEqualsWorker(MethodTable* mt)
+{
+    MethodTable::IntroducedMethodIterator it(mt);
+    for (; it.IsValid(); it.Next())
+    {
+        if (strcmp(it.GetMethodDesc()->GetName(), "Equals") == 0)
```

Comment:
> The class may have an Equals method that does not override Object.Equals
Umm. I may need to check if the Equals is from overriding (if it can be figured out).

>  In MSIL it is also possible to override a method using a different method name.
I think this is out of scope of this issue. If we need to care about this case, we may need to fix from another way - may be generating different IL code for Equals? @jkotas